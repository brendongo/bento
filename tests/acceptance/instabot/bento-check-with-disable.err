Running Bento checks...

⚠ 25 finding(s) in  s
◦ To suppress all findings, run $ bento archive.
