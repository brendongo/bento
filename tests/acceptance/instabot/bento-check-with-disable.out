bandit start-process-with-a-shell https://bandit.readthedocs.io/en/latest/plugins/b605_start_process_with_a_shell.html
     > examples/multi_script_CLI.py:42
     ╷
   42│   os.system("cls")
     ╵
     = Starting a process with a shell: Seems safe, but may be changed in the
       future, consider rewriting without shell

bandit try-except-pass https://bandit.readthedocs.io/en/latest/plugins/b110_try_except_pass.html
     > instabot/api/api.py:538
     ╷
  538│   except Exception:
     ╵
     = Try, Except, Pass detected.

flake8 no-mutable-default-args https://github.com/PyCQA/flake8-bugbear/blob/master/README.rst#list-of-warnings
     > instabot/api/api.py:673
     ╷
  673│   def get_timeline_feed(self, options=[]):
     ╵
     = Do not use mutable data structures for argument defaults.  They are
       created during function definition time. All calls to the function reuse
       this one instance of that data structure, persisting changes between
       them.

flake8 no-mutable-default-args https://github.com/PyCQA/flake8-bugbear/blob/master/README.rst#list-of-warnings
     > instabot/api/api.py:736
     ╷
  736│   options={},
     ╵
     = Do not use mutable data structures for argument defaults.  They are
       created during function definition time. All calls to the function reuse
       this one instance of that data structure, persisting changes between
       them.

flake8 no-mutable-default-args https://github.com/PyCQA/flake8-bugbear/blob/master/README.rst#list-of-warnings
     > instabot/api/api.py:777
     ╷
  777│   self, video, caption=None, upload_id=None, thumbnail=None, options={}
     ╵
     = Do not use mutable data structures for argument defaults.  They are
       created during function definition time. All calls to the function reuse
       this one instance of that data structure, persisting changes between
       them.

flake8 no-mutable-default-args https://github.com/PyCQA/flake8-bugbear/blob/master/README.rst#list-of-warnings
     > instabot/api/api.py:817
     ╷
  817│   options={},
     ╵
     = Do not use mutable data structures for argument defaults.  They are
       created during function definition time. All calls to the function reuse
       this one instance of that data structure, persisting changes between
       them.

flake8 no-mutable-default-args https://github.com/PyCQA/flake8-bugbear/blob/master/README.rst#list-of-warnings
     > instabot/api/api_photo.py:107
     ╷
  107│   options={},
     ╵
     = Do not use mutable data structures for argument defaults.  They are
       created during function definition time. All calls to the function reuse
       this one instance of that data structure, persisting changes between
       them.

flake8 no-mutable-default-args https://github.com/PyCQA/flake8-bugbear/blob/master/README.rst#list-of-warnings
     > instabot/api/api_video.py:102
     ╷
  102│   options={}
     ╵
     = Do not use mutable data structures for argument defaults.  They are
       created during function definition time. All calls to the function reuse
       this one instance of that data structure, persisting changes between
       them.

flake8 no-mutable-default-args https://github.com/PyCQA/flake8-bugbear/blob/master/README.rst#list-of-warnings
     > instabot/api/api_video.py:234
     ╷
  234│   options={}
     ╵
     = Do not use mutable data structures for argument defaults.  They are
       created during function definition time. All calls to the function reuse
       this one instance of that data structure, persisting changes between
       them.

flake8 no-mutable-default-args https://github.com/PyCQA/flake8-bugbear/blob/master/README.rst#list-of-warnings
     > instabot/bot/bot.py:170
     ╷
  170│   blacklist_hashtags=["#shop", "#store", "#free"],
     ╵
     = Do not use mutable data structures for argument defaults.  They are
       created during function definition time. All calls to the function reuse
       this one instance of that data structure, persisting changes between
       them.

flake8 no-mutable-default-args https://github.com/PyCQA/flake8-bugbear/blob/master/README.rst#list-of-warnings
     > instabot/bot/bot.py:752
     ╷
  752│   options={}
     ╵
     = Do not use mutable data structures for argument defaults.  They are
       created during function definition time. All calls to the function reuse
       this one instance of that data structure, persisting changes between
       them.

flake8 no-mutable-default-args https://github.com/PyCQA/flake8-bugbear/blob/master/README.rst#list-of-warnings
     > instabot/bot/bot.py:780
     ╷
  780│   def upload_video(self, video, caption="", thumbnail=None, options={}):
     ╵
     = Do not use mutable data structures for argument defaults.  They are
       created during function definition time. All calls to the function reuse
       this one instance of that data structure, persisting changes between
       them.

bandit pickle-used https://bandit.readthedocs.io/en/latest/blacklists/blacklist_calls.html#b301-pickle
     > instabot/bot/bot_checkpoint.py:66
     ╷
   66│   checkpoint = pickle.load(f)
     ╵
     = Pickle and modules that wrap it can be unsafe when used to deserialize
       untrusted data, possible security issue.

bandit try-except-pass https://bandit.readthedocs.io/en/latest/plugins/b110_try_except_pass.html
     > instabot/bot/bot_checkpoint.py:71
     ╷
   71│   except Exception:
     ╵
     = Try, Except, Pass detected.

flake8 no-mutable-default-args https://github.com/PyCQA/flake8-bugbear/blob/master/README.rst#list-of-warnings
     > instabot/bot/bot_photo.py:13
     ╷
   13│   options={}
     ╵
     = Do not use mutable data structures for argument defaults.  They are
       created during function definition time. All calls to the function reuse
       this one instance of that data structure, persisting changes between
       them.

flake8 no-mutable-default-args https://github.com/PyCQA/flake8-bugbear/blob/master/README.rst#list-of-warnings
     > instabot/bot/bot_video.py:4
     ╷
    4│   def upload_video(self, video, caption="", thumbnail=None, options={}):
     ╵
     = Do not use mutable data structures for argument defaults.  They are
       created during function definition time. All calls to the function reuse
       this one instance of that data structure, persisting changes between
       them.

shellcheck SC2148 https://github.com/koalaman/shellcheck/wiki/SC2148
     > push.sh:1
     ╷
    1│   LAST_COMMIT_MESSAGE="$(git log --no-merges -2 --pretty=%B)"
     ╵
     = Tips depend on target shell and yours is unknown. Add a shebang.

shellcheck SC2086 https://github.com/koalaman/shellcheck/wiki/SC2086
     > push.sh:6
     ╷
    6│   git remote add origin https://${GITHUB_TOKEN}@github.com/instagrambot/instabot.git
     ╵
     = Double quote to prevent globbing and word splitting.

shellcheck SC2086 https://github.com/koalaman/shellcheck/wiki/SC2086
     > push.sh:13
     ╷
   13│   twine upload -u $PYPI_USERNAME -p $PYPI_PASSWORD dist/*
     ╵
     = Double quote to prevent globbing and word splitting.

shellcheck SC2086 https://github.com/koalaman/shellcheck/wiki/SC2086
     > push.sh:13
     ╷
   13│   twine upload -u $PYPI_USERNAME -p $PYPI_PASSWORD dist/*
     ╵
     = Double quote to prevent globbing and word splitting.

